pkg/pb/database.go

```diff
@@ -25,141 +25,141 @@ import (
 	v1 "github.com/apache/skywalking-banyandb/api/proto/banyandb/v1"
 )
 
-type shardEventBuilder struct {
+type ShardEventBuilder struct {
```

Comment:
The exported func in this file return unexported types, which causes the lint check to fail. @lujiajing1126 Would you pls check whether the way I fixed is correct?

Reply:
> The exported func in this file return unexported types, which causes the lint check to fail. @lujiajing1126 Would you pls check whether the way I fixed is correct?

Yes. I suppose the change is correct. I intended not to expose/export the structs and only allow users to create `builder`(s) by the given functions.

I don't know why this pattern is not recommended by the linter. Ref: https://stackoverflow.com/a/21470517

Reply:
It's not a common pattern in go. If you want to force users to use the factory method, an interface should be better, which is a canonical pattern to gophers.

Reply:
> It's not a common pattern in go. If you want to force users to use the factory method, an interface should be better, which is a canonical pattern to gophers.

Sure. Make sense to me.

Reply:
> It's not a common pattern in go. If you want to force users to use the factory method, an interface should be better, which is a canonical pattern to gophers.

I did some investigation. I seems this pattern does commonly used by developers.
See https://github.com/golang/lint/issues/210